x-pack/legacy/plugins/apm/server/lib/service_map/setup_required_scripts.ts

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Legacy } from 'kibana';
+import { CallCluster } from 'src/legacy/core_plugins/elasticsearch';
+
+const MAP_SERVICE_CONNS_SCRIPT_ID = 'map-service-conns';
+const REDUCE_SERVICE_CONNS_SCRIPT_ID = 'reduce-service-conns';
+const COMBINE_SERVICE_CONNS_SCRIPT_ID = 'combine-service-conns';
+const EXTRACT_DESTINATION_INGEST_PIPELINE_ID = 'extract_destination';
+const APM_INGEST_PIPELINE_ID = 'apm';
+
+async function putScriptMapServiceConns(callCluster: CallCluster) {
+  return await callCluster('putScript', {
+    id: MAP_SERVICE_CONNS_SCRIPT_ID,
+    body: {
+      script: {
+        lang: 'painless',
+        source: `
+          def s = new HashMap();
+
+          if (!doc['span.id'].empty) {
+            s.id = doc['span.id'].value
+          } else {
+            s.id = doc['transaction.id'].value;
+            s.transaction = true;
+          }
+          if (!doc['parent.id'].empty) {
+            s.parent = doc['parent.id'].value;
+          }
+          if (!doc['service.environment'].empty) {
+            s.environment = doc['service.environment'].value;
+          }
+
+          if (!doc['destination.address'].empty) {
+            s.destination = doc['destination.address'].value;
+          }
+
+          if (!doc['_index'].empty) {
+            s._index = doc['_index'].value;
+          }
+
+          if (!doc['span.type'].empty) {
+            s.span_type = doc['span.type'].value;
+          }
+
+          if (!doc['span.subtype'].empty) {
+            s.span_subtype = doc['span.subtype'].value;
+          }
+
+          s.timestamp = doc['@timestamp'].value;
+          s.service_name = doc['service.name'].value;
+          if(!state.spans.containsKey(s.parent)) {
+            state.spans.put(s.parent, new ArrayList())
+          }
+
+          if (s.parent != s.id) {
+            state.spans[s.parent].add(s)
+          }
+        `
+      }
+    }
+  });
+}
+
+async function putScriptReduceServiceConns(callCluster: CallCluster) {
+  return await callCluster('putScript', {
+    id: REDUCE_SERVICE_CONNS_SCRIPT_ID,
+    body: {
+      script: {
+        lang: 'painless',
+        source: `
+          void extractChildren(def caller, def spans, def upstream, def conns, def count) {
+            // TODO: simplify this
+            if (spans.containsKey(caller.id)) {
+              for(s in spans[caller.id]) {
+                  if (caller.span_type=='external') {
```

Comment:
I see `external` being used a couple of places. Should it be a constant like we have for [transaction types](https://github.com/elastic/kibana/blob/45ef370e8443baf7eeec5bc2cb7e25cae31fff10/x-pack/legacy/plugins/apm/common/transaction_types.ts#L9)?